src/main/client/admin.c

```diff
@@ -31,124 +31,165 @@
-    static char *kwlist[] = {"user", "password", "roles", "policy", NULL};
-
-    // Python Function Argument Parsing
-    if (PyArg_ParseTupleAndKeywords(args, kwds, "OOO|O:admin_create_user",
-                                    kwlist, &py_user, &py_password, &py_roles,
-                                    &py_policy) == false) {
-        return NULL;
-    }
-
-    // Aerospike Operation Arguments
-    int roles_size = 0;
-    char **roles = NULL;
-    const char *user, *password;
-
     if (!self || !self->as) {
         as_error_update(&err, AEROSPIKE_ERR_PARAM, "Invalid aerospike object");
-        goto CLEANUP;
+        goto RAISE_EXCEPTION;
     }
 
     if (!self->is_conn_16) {
         as_error_update(&err, AEROSPIKE_ERR_CLUSTER,
                         "No connection to aerospike cluster");
-        goto CLEANUP;
+        goto RAISE_EXCEPTION;
     }
 
     // Convert python object to an array of roles
-    if (PyList_Check(py_roles)) {
-        roles_size = PyList_Size(py_roles);
-        roles = alloca(sizeof(char *) * roles_size);
-        for (int i = 0; i < roles_size; i++) {
-            roles[i] = cf_malloc(sizeof(char) * AS_ROLE_SIZE);
-            memset(roles[i], 0, sizeof(char) * AS_ROLE_SIZE);
-        }
-    }
-    else {
+    if (!PyList_Check(py_roles)) {
         as_error_update(&err, AEROSPIKE_ERR_PARAM, "Roles should be a list");
-        goto CLEANUP;
+        goto RAISE_EXCEPTION;
+    }
+    int roles_size = PyList_Size(py_roles);
+    if (PyErr_Occurred()) {
+        goto RAISE_EXCEPTION;
+    }
+    char **roles = alloca(sizeof(char *) * roles_size);
+    for (int i = 0; i < roles_size; i++) {
+        roles[i] = cf_malloc(sizeof(char) * AS_ROLE_SIZE);
+        memset(roles[i], 0, sizeof(char) * AS_ROLE_SIZE);
     }
 
     pyobject_to_strArray(&err, py_roles, roles, AS_ROLE_SIZE);
     if (err.code != AEROSPIKE_OK) {
-        goto CLEANUP;
+        goto CLEANUP_AND_RAISE_EXCEPTION;
     }
 
     // Convert python objects to username and password strings
-    if (!PyUnicode_Check(py_user)) {
-        as_error_update(&err, AEROSPIKE_ERR_PARAM,
-                        "Username should be a string");
-        goto CLEANUP;
-    }
-
-    user = PyUnicode_AsUTF8(py_user);
-
-    if (!PyUnicode_Check(py_password)) {
-        as_error_update(&err, AEROSPIKE_ERR_PARAM,
-                        "Password should be a string");
-        goto CLEANUP;
+    const char *user = convert_pyobject_to_str(py_user);
+    if (!user) {
+        as_error_update(&err, AEROSPIKE_ERR_PARAM, STR_CONVERSION_ERROR_MSG,
+                        "Username");
+        goto CLEANUP_AND_RAISE_EXCEPTION;
+    }
+
+    const char *password = NULL;
+    if (py_password) {
+        password = convert_pyobject_to_str(py_password);
+        if (!password) {
+            as_error_update(&err, AEROSPIKE_ERR_PARAM, STR_CONVERSION_ERROR_MSG,
+                            "Password");
+            goto CLEANUP_AND_RAISE_EXCEPTION;
+        }
```

Comment:
```suggestion
    const char *user = convert_pyobject_to_str(py_user);
    if (!user) {
        as_error_update(&err, AEROSPIKE_ERR_PARAM, STR_CONVERSION_ERROR_MSG,
                        "username");
        goto CLEANUP_AND_RAISE_EXCEPTION;
    }

    const char *password = NULL;
    if (py_password) {
        password = convert_pyobject_to_str(py_password);
        if (!password) {
            as_error_update(&err, AEROSPIKE_ERR_PARAM, STR_CONVERSION_ERROR_MSG,
                            "password");
            goto CLEANUP_AND_RAISE_EXCEPTION;
        }
```
IMO the name of the parameter in the error message should match the parameter name in code. Also, not sure if it should be `username` or `user` since the docs say `username` but the keywords in the code below say `user`.

Reply:
I'm fine with what you decide 👍